nsxt/resource_nsxt_vpc_gateway_policy.go

```diff
@@ -0,0 +1,163 @@
+/* Copyright © 2024 Broadcom, Inc. All Rights Reserved.
+   SPDX-License-Identifier: MPL-2.0 */
+
+package nsxt
+
+import (
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/vmware/vsphere-automation-sdk-go/runtime/bindings"
+	"github.com/vmware/vsphere-automation-sdk-go/runtime/data"
+	"github.com/vmware/vsphere-automation-sdk-go/services/nsxt/model"
+
+	"github.com/vmware/terraform-provider-nsxt/api/infra/domains"
+	utl "github.com/vmware/terraform-provider-nsxt/api/utl"
+)
+
+func resourceNsxtVPCGatewayPolicy() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceNsxtVPCGatewayPolicyCreate,
+		Read:   resourceNsxtVPCGatewayPolicyRead,
+		Update: resourceNsxtVPCGatewayPolicyUpdate,
+		Delete: resourceNsxtVPCGatewayPolicyDelete,
+		Importer: &schema.ResourceImporter{
+			State: nsxtPolicyPathResourceImporter,
+		},
+
+		Schema: getPolicyGatewayPolicySchema(false),
+	}
+}
+
+func resourceNsxtVPCGatewayPolicyCreate(d *schema.ResourceData, m interface{}) error {
+	connector := getPolicyConnector(m)
+
+	// Initialize resource Id and verify this ID is not yet used
+	id, err := getOrGenerateID2(d, m, resourceNsxtPolicyGatewayPolicyExistsPartial(""))
+	if err != nil {
+		return err
+	}
+
+	err = policyGatewayPolicyBuildAndPatch(d, m, connector, isPolicyGlobalManager(m), id, false)
+	if err != nil {
+		return handleCreateError("VPC Gateway Policy", id, err)
+	}
+
+	d.SetId(id)
+	d.Set("nsx_id", id)
+
+	return resourceNsxtVPCGatewayPolicyRead(d, m)
+}
+
+func resourceNsxtVPCGatewayPolicyRead(d *schema.ResourceData, m interface{}) error {
+	connector := getPolicyConnector(m)
+
+	id := d.Id()
+	if id == "" {
+		return fmt.Errorf("error obtaining VPC Gateway Policy ID")
+	}
+
+	obj, err := getGatewayPolicyInDomain(getSessionContext(d, m), id, "", connector)
```

Comment:
I understand the reluctance to rename common helpers, but I personally find `domain` confusing here.. I think a comment explaining that we're not actually looking in domain for VPC would help future code readers. 
For a long run, I think we might want to consider moving `domain` into the context

Reply:
I think that moving domain (or generally anything that's in the URL part of the API) into context makes some sense, but I'd rather avoid this now - it's a major effort and a compatibility issue.
For the func name change - I'll change it, I missed that somehow.